File: src/arm_gazebo/scripts/read_depth_image.py
#!/usr/bin/env python

import rospy
from sensor_msgs.msg import Image
from cv_bridge import CvBridge, CvBridgeError
import cv2

def depth_image_callback(msg):
    bridge = CvBridge()
    
    try:
        # Convert the ROS image message to an OpenCV image (float32 encoding for depth in meters)
        depth_image = bridge.imgmsg_to_cv2(msg, desired_encoding="32FC1")
    except CvBridgeError as e:
        rospy.logerr("CvBridge Error: {0}".format(e))
        return
    
    # Display or process the depth image
    cv2.imshow("Depth Image", depth_image)
    cv2.waitKey(1)

def main():
    rospy.init_node('depth_image_subscriber')

    # Subscribe to the depth image topic
    rospy.Subscriber("/camera/depth/image_raw", Image, depth_image_callback)

    rospy.spin()

    cv2.destroyAllWindows()

if __name__ == "__main__":
    main()

File: src/pick_and_place/src/read_depth_image.cpp
#include <ros/ros.h>
#include <sensor_msgs/Image.h>
#include <cv_bridge/cv_bridge.h>
#include <opencv2/opencv.hpp>

void depthImageCallback(const sensor_msgs::ImageConstPtr& msg) {
    try {
        // Convert the ROS image message to an OpenCV image (CV_32FC1 for depth in meters)
        cv::Mat depth_image = cv_bridge::toCvShare(msg, "32FC1")->image;

        // Display the depth image
        cv::imshow("Depth Image", depth_image);
        cv::waitKey(1);
    } catch (cv_bridge::Exception& e) {
        ROS_ERROR("CvBridge Error: %s", e.what());
    }
}

int main(int argc, char** argv) {
    // Initialize the ROS node
    ros::init(argc, argv, "depth_image_subscriber");

    ros::NodeHandle nh;

    // Subscribe to the depth image topic
    ros::Subscriber sub = nh.subscribe("/camera/depth/image_raw", 10, depthImageCallback);

    // Start the ROS event loop
    ros::spin();

    // Close OpenCV windows when the node exits
    cv::destroyAllWindows();

    return 0;
}
